machine-learning/src/main/java/com/baeldung/cnn/MNISTDataSet.java

```diff
@@ -0,0 +1,63 @@
+package com.baeldung.cnn;
+
+import org.datavec.api.records.reader.impl.collection.ListStringRecordReader;
+import org.datavec.api.split.ListStringSplit;
+import org.deeplearning4j.datasets.datavec.RecordReaderDataSetIterator;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static java.nio.file.Files.readAllBytes;
+import static java.util.stream.Collectors.toList;
+
+class MNISTDataSet {
+    public static final String TRAIN_LABELS = "src/main/resources/mnist/train-labels.idx1-ubyte";
+    public static final String TRAIN_IMAGES = "src/main/resources/mnist/train-images.idx3-ubyte";
+    public static final String TEST_LABELS = "src/main/resources/mnist/t10k-labels.idx1-ubyte";
+    public static final String TEST_IMAGES = "src/main/resources/mnist/t10k-images.idx3-ubyte";
+    private static int OFFSET_SIZE = 4;
+    private static int NUM_ITEMS_OFFSET = 4;
+    private static int ITEMS_SIZE = 4;
+    private static int ROWS = 28;
+    private static int COLUMNS = 28;
+    private static int IMAGE_OFFSET = 16;
+    private static int IMAGE_SIZE = ROWS * COLUMNS;
+
+    public static RecordReaderDataSetIterator load(boolean test) throws IOException, InterruptedException {

```

Comment:
In general, it can be cleaner to provide three functions:

- `loadTest()`
- `loadTraining()`
- `load(File, File) // called by the other two`

This avoids the function that switches its meaning by an incoming boolean...

Reply:
Indeed, this is a very good catch. Solved it and pushed.